### PR TITLE
Update gradle

### PR DESCRIPTION
**Pull request by fada21**
## Problem 

All4 project needs update of gradle to 6.0.1. Currently published artefact of this library couldn't be consumed in all4. The issue seemed to be `GRADLE_METADATA)` feature. 
```
 ./gradlew test
Configuration on demand is an incubating feature.

> Configure project :aws-v4-signer-java:aws-v4-signer
Kotlin Multiplatform Projects are an experimental feature.
This build is set up to publish Kotlin multiplatform libraries with experimental Gradle metadata. Future Gradle versions may fail to resolve dependencies on these publications. You can disable Gradle metadata usage during publishing and dependencies resolution by removing enableFeaturePreview('GRADLE_METADATA')
```

## Solution

Update gradle to 6.0.1 for this library as well with all necessary adjustments. Remove `enableFeaturePreview('GRADLE_METADATA')` to allow gradle 6 publish it's required. 

## Tests

None added. Updated some annotations to address warning for few.

## Paired with

@pablisco

### CODE REVIEW

**Comment by fada21** [comment]
Getting local maven version of published library fixes all4 project with gradle updated to 6.0.1

File: build.gradle [hunk 1]
```diff
@@ -1,12 +1,14 @@
 buildscript {
-    ext.kotlin_version = '1.3.20'
-    ext.ktor_version = '1.1.2'
+    ext.kotlin_version = '1.3.61'
+    ext.ktor_version = '1.2.6'
+    ext.serialization_version = '0.14.0'
```
Comment (fada21) [inline build.gradle:4]: Some extension method were using those apis but after other libs update above one needed explicit version as it was failing the build.

File: build.gradle [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
     repositories {
         jcenter()
         maven { url 'https://dl.bintray.com/jetbrains/kotlin-native-dependencies' }
     }
     dependencies {
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+        classpath "org.jetbrains.kotlin:kotlin-serialization:$kotlin_version"
```
Comment (JozefCeluch) [inline build.gradle:11]: how come we're using the `kotlin_version` here when we have `serialization_version`?

**Review by JozefCeluch — APPROVED** [review]

**Review by logicalusman — APPROVED** [review]

File: build.gradle (hunk 2 above)
Comment (fada21) [reply build.gradle:11]: that's what jetbrains have in manual:
https://github.com/Kotlin/kotlinx.serialization#gradle

and then for `kotlinx-serialization-runtime`
https://github.com/Kotlin/kotlinx.serialization#mavenjvm

**Pull request merged**